src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedParameterSymbol.cs

```diff
@@ -36,6 +37,11 @@ internal abstract class SynthesizedParameterSymbolBase : ParameterSymbol
             _ordinal = ordinal;
             _refKind = refKind;
             _name = name;
+
+            if (this.RefKind == RefKind.RefReadOnly)
+            {
+                this.DeclaringCompilation.EnsureIsReadOnlyAttributeExists(this);
```

Comment:
>this.DeclaringCompilation.EnsureIsReadOnlyAttributeExists(this); [](start = 16, length = 64)

It is not obvious whether the diagnostics should go to the compilation diagnostics bag.